satellite/metainfo/validation.go

```diff
@@ -369,5 +370,20 @@ func (endpoint *Endpoint) validatePieceHash(ctx context.Context, piece *pb.Remot
 			return errs.New("piece hash PieceSize is larger than order limit, removing from pointer (%v > %v)", piece.Hash.PieceSize, limit.Limit)
 		}
 	}
+
+	publicKey, err := endpoint.certdb.GetPublicKey(ctx, piece.NodeId)
```

Comment:
Note, we always load it from the database otherwise it'll add massive amount of traffic to the database.

Similarly there should be one DB call per verification rather than ~100.